cmd/scollector/collectors/collectors.go

```diff
@@ -325,3 +325,17 @@ func skipMetric(index string) bool {
 	}
 	return true
 }
+
+type tsIntegrator func(int64, float64) float64
+
+func getTsIntegrator() tsIntegrator {
+	var i float64
```

Comment:
I was thinking there might be use cases where you don't want to use it in
the map, so this would make it more reusable

On Thu, Dec 3, 2015 at 10:23 AM, Craig Peterson <notifications@github.com>
wrote:

> In cmd/scollector/collectors/collectors.go
> <https://github.com/bosun-monitor/bosun/pull/1494#discussion_r46564464>:
>
> > @@ -325,3 +325,17 @@ func skipMetric(index string) bool {
> >  	}
> >  	return true
> >  }
> > +
> > +type tsIntegrator func(int64, float64) float64
> > +
> > +func getTsIntegrator() tsIntegrator {
> > +	var i float64
>
> How about
>
> type tsIntegrator struct{
>    data map[string]struct{
>       total floats64
>       lastTime
>    }
>    sync.Mutex
> }
>
> ?
>
> then you can do cpuIntegrator.Add(key, now, val) and
> cpuIntegrator.Total(key). Hide all of the mutexing and map stuff inside
> the integrator.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/bosun-monitor/bosun/pull/1494/files#r46564464>.
>